src/main/java/edu/ie3/io/processor/EntityProcessor.java

```diff
@@ -0,0 +1,214 @@
+                }
+              });
+
+    } catch (Exception e) {
+      throw new EntityProcessorException(
+          "Error during EntityProcessor class registration process. Exception was:" + e);
+    }
+
+    // uuid should always be the first element in the map
+    String[] filteredArray =
+        fieldNameToMethod.keySet().stream()
+            .filter(x -> !x.toLowerCase().contains(uuidString))
+            .toArray(String[]::new);
+    return ArrayUtils.addAll(new String[] {uuidString}, filteredArray);
+  }
+
+  /**
+   * Standard call to handle an entity
+   *
+   * @param entity the entity that should be 'de-serialized' into a map of fieldName -> fieldValue
+   * @return an optional Map with fieldName -> fieldValue or an empty optional if an error occurred
+   *     during processing
+   */
+  public Optional<LinkedHashMap<String, String>> handleEntity(T entity) {
+    if (!registeredClass.equals(entity.getClass()))
+      throw new FactoryException(
+          "Cannot process "
+              + entity.getClass().getSimpleName()
+              + ".class with this EntityProcessor. Please either provide an element of "
+              + registeredClass.getSimpleName()
+              + ".class or create a new factory for "
+              + entity.getClass().getSimpleName()
+              + ".class!");
+    return processEntity(entity);
+  }
+
+  /**
+   * Actual implementation of the handling process. Depends on the entity that should be processed
+   * and hence needs to be implemented individually
+   *
+   * @param entity the entity that should be 'de-serialized' into a map of fieldName -> fieldValue
+   * @return an optional Map with fieldName -> fieldValue or an empty optional if an error occurred
+   *     during processing
+   */
+  protected abstract Optional<LinkedHashMap<String, String>> processEntity(T entity);
+
+  /**
+   * Standard method to process a ZonedDateTime to a String based on a method return object NOTE:
+   * this method does NOT check if the provided object is of type ZonedDateTime. This has to be done
+   * manually BEFORE calling this method!
+   *
+   * @param zonedDateTime representation of the ZonedDateTime
+   * @return string representation of the ZonedDateTime
+   */
+  protected String processZonedDateTime(ZonedDateTime zonedDateTime) {
+    return TimeTools.toString(zonedDateTime);
+  }
+
+  /**
+   * Standard method to process a Quantity to a String based on a method return object
+   *
+   * @param quantity the quantity that should be processed
+   * @param fieldName the field name the quantity is set to
+   * @param resultModel true if the processed model is of type ResultEntity, false otherwise
+   * @return an optional string with the normalized to {@link StandardUnits} value of the quantity
+   *     or empty if an error occurred during processing
+   */
+  protected Optional<String> handleQuantity(
+      Quantity<?> quantity, String fieldName, boolean resultModel) {
```

Comment:
I feel like handleQuantity should just be specialized in ResultEntityProcessor? Then resultModel boolean could be dropped.

Reply:
I agree on this. Restructured the code accordingly and moved the special handling into `ResultEntityProcessor` while keeping the generalized method and avoid code duplicates. Please let me know if you think this is a valid way to go.